core/src/solution_submission.rs

```diff
@@ -108,20 +107,31 @@ impl<'a> StableXSolutionSubmitter<'a> {
         &self,
         batch_index: u32,
         solution: Solution,
-        err: MethodError,
+        err: RetryError,
     ) -> SolutionSubmissionError {
-        if let Some(tx) = extract_transaction_receipt(&err) {
-            if let Some(block_number) = tx.block_number {
-                if let Err(err) = self
-                    .contract
-                    .get_solution_objective_value(batch_index, solution, Some(block_number.into()))
-                    .await
-                {
-                    return SolutionSubmissionError::from(err);
+        match err {
+            RetryError::MethodError(err) => {
+                if let Some(tx) = extract_transaction_receipt(&err) {
+                    if let Some(block_number) = tx.block_number {
+                        if let Err(err) = self
+                            .contract
+                            .get_solution_objective_value(
+                                batch_index,
+                                solution,
+                                Some(block_number.into()),
+                            )
+                            .await
+                        {
+                            return SolutionSubmissionError::from(err);
+                        }
+                    }
                 }
+                SolutionSubmissionError::Unexpected(err.into())
+            }
+            RetryError::TransactionNotConfirmedInTime => {
+                SolutionSubmissionError::Benign("transaction not confirmed in time".to_string())
```

Comment:
Should this be a benign error?

Reply:
I would say this is a bad error, since in this case we are not guaranteed that some solver provided a solution. Thus we should log this as a failure and work towards getting those cases to 0 (we can consider not alerting on them if they are frequent, but they should be measured somewhere).